lib/mix/lib/mix/tasks/test.coverage.ex

```diff
@@ -278,10 +286,8 @@ defmodule Mix.Tasks.Test.Coverage do
   end
 
   defp display({percentage, name}, opts) do
-    threshold = Keyword.get(opts, :threshold, @default_threshold)
-
     Mix.shell().info([
-      color(percentage, threshold),
+      color(percentage, threshold(opts)),
```

Comment:
Let's compute the threshold before we call `display` and pass it as an argument to `display`.